conda/core/subdir_data.py

```diff
@@ -87,9 +87,14 @@ def query_all(package_ref_or_match_spec, channels=None, subdirs=None,
             package_ref_or_match_spec))
 
         # TODO test timing with ProcessPoolExecutor
-        with ThreadLimitedThreadPoolExecutor() as executor:
-            result = tuple(concat(executor.map(subdir_query, channel_urls)))
-        return result
+        if any('s3://' in url for url in channel_urls):
```

Comment:
fixes #8993 maybe

Reply:
You could probably allow running s3 checks in parallel if you added `_ = boto3.client('s3')` to `gateways/connection/adapters/s3.py` to allow the first time initialization to happen on main thread. Subsequent calls should be thread-safe. This would require a bit of testing?

Reply:
Please submit a PR if you can.  I'm not very familiar with s3 at all, and don't have confidence in my ability to write meaningful tests.

Reply:
I gave it a go here: https://github.com/conda/conda/pull/9037

Reply:
Thanks!  I'm going to just revert this blob, as I think your PR is a better approach.